build_tools/docker/manylinux2014_x86_64-release/Dockerfile

```diff
@@ -50,3 +50,16 @@ ENV BAZEL_LINKLIBS "-lstdc++ -lm"
 # fetch/build/install exactly what is needed.
 COPY install_tbb_manylinux2014.sh /usr/local/bin
 RUN bash /usr/local/bin/install_tbb_manylinux2014.sh
+
+######## AMD ROCM #######
```

Comment:
I don't necessarily object to installing more dependencies in Docker images, but I think we should have a conversation about the "experimental" part of the ROCm driver and the not-experimental part of the releases.

If the goal is to set `-DIREE_EXTERNAL_HAL_DRIVERS=rocm` in the official release builds, then expect some push-back. Ideally, we'd promote the driver from experimental status - move the code, finish missing features, enable builds and tests on presubmit, etc. . It's not going to be enough to just set that option in _only_ the release builds. We need to be testing for build failures and such much more directly, so we don't get unexpected release build failures that are difficult to investigate.

If the goal is to build a downstream release or upstream "unstable/experimental features" release, then we could skip a few of those steps (a downstream release could also inherit / fork this Dockerfile).

Reply:
I think we are trying to move it out of the "experimental" category and are willing to submit PRs for it. However we will need to bringup the ROCM hardware so we can run pre-submit etc. 

Until we get the hardware in place having this dependency here allows inheriting this downstream without forking the Docker images too (since we gradually want to see all these changes back in IREE). 

So would it be ok to land this so we have one thing out of the way -- rocm deps. Then we enable ROCM downstream until we can do so in the official build (with tests etc).

Reply:
I agree with Scott. I'm ok adding this if it doesn't explode the image size or time to build. It sounds like what you want to do is be able to use this but not immediately enable ROCM in the release builds, which seems fine. Another option though would be to use a docker image that inherits from this one. That would also give you the freedom to build it yourself :-P

Reply:
In that case, how about also adding deps to IREE's base Dockerfile too? (the release image does not inherit from base, but all the others do)

You'd add a block similar to these:

https://github.com/iree-org/iree/blob/2f48a9589ee86f61a03c822d7be7d44a6d244edf/build_tools/docker/base/Dockerfile#L133-L162

Reply:
If we're making analogy between rocm and cuda, this should not go in the base image, but rather in a separate -amd image (like we have a -nvidia image). The main reasons for that, IIRC, were that it was a big dependency and that having it in a system without a real GPU really confused things and made it difficult to have swiftshader in there.

Reply:
I think having it in the IREE base or -amd image makes sense.   Since we wont have the swiftshader issue I'll add it to the same block @ScottTodd pointed to. 

About inheriting / forking the docker image -- just trying to reduce the spread of the forks downstream so there is incentive to incrementally add / move the ROCM backend towards full fledged support.

Reply:
We need to differentiate between dependencies needed to *build*, which (assuming it isn't "too big"), I am supportive of just including in the main system building images (i.e. base or its successor and/or the manylinux image). It is a massive pita to not have build dependencies in common.

I'm also ok enabling the ROCM backend in our releases (although if truly experimental, would rather the compiler backend be "experimental-rocm" or something until stable). However, it has to be enabled to at least build in presubmits before enabling building in releases.

The final tier of leaving experimental status is test bots. We're not there yet, and we can cross that bridge later. In general, though, it is better for everyone if active/known/primary backends (ROCM is one) are built regularly and are part of released tools. That gives us the ability to ease into better support without fork the world situations. I know I'm defining a category here without rigor, but suffice to say that ROCM is in and "MyStartupThatNoOneHasHeardOf Uber GPU" is not.

Reply:
Yup I agree with everything you're saying too 🙂 I'm just trying to understand what we're trying to accomplish here. GPU things tend to big and then interfere with other GPU things (which is why cuda is in a separate image, for instance), but if this is small, I don't mind adding it wherever makes sense

Reply:
updated in latest patch